repository_service_tuf_worker/repository.py

```diff
@@ -1018,6 +1104,67 @@ def _update_metadata_delegation(
 
         return (success, failed)
 
+    def _update_delegated_role_target_files(self, delegation, db_role):
+        delegation.signed.targets.clear()
+        delegation.signed.targets = {
+            file.path: TargetFile.from_dict(file.info, file.path)
+            for file in db_role.target_files
+            if file.action == targets_schema.TargetAction.ADD
+            # Filtering the files with action 'ADD' cannot be done
+            # in CRUD. If a target role doesn't have any target
+            # files with an action 'ADD' (only 'REMOVE') then using
+            # CRUD will not return the target role and it won't be
+            # updated. An example can be when there is a role with
```

Comment:
But don't you do that here as well?
Now, when you filter by `if file.action == targets_schema.TargetAction.ADD`?

Reply:
First, this is a statement copy from here
https://github.com/repository-service-tuf/repository-service-tuf-worker/blob/e5533ac15120a27cc4a93582930f04ddc7dd837f/repository_service_tuf_worker/repository.py#L457

What it means is that I could do CRUD filtering the db_roles object with action == 'ADD'
But there is a problem the DB query can return None the role as only 'REMOVED' and the delegated role (without any artifact ADD in action) will remain outdated.

We had discussed it here https://github.com/repository-service-tuf/repository-service-tuf-worker/pull/289#discussion_r1198873147 (few years ago)

The goal idea is to refactor the `_update_snapshot`